File: scala/sdktest/OutputRegisterUserEntityTest.scala
// Generated basic-flow test for the output_register_user entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped OutputRegisterUserTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.bluefintecsuserbackofficesdk.core.{Helpers, BluefinTecsUserBackofficeSDK}
import voxgig.bluefintecsuserbackofficesdk.utility.struct.Struct

object OutputRegisterUserEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("output_register_user.instance") {
      val testsdk = BluefinTecsUserBackofficeSDK.testSDK()
      val ent = testsdk.outputRegisterUser(null)
      rep.check("output_register_user.instance", ent != null, "expected non-null output_register_user entity")
    }

    rep.scope("output_register_user.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/output_register_user/OutputRegisterUserTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = BluefinTecsUserBackofficeSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("output_register_user01", "OUTPUT_REGISTER_USER01")
      idmap.put("output_register_user02", "OUTPUT_REGISTER_USER02")
      idmap.put("output_register_user03", "OUTPUT_REGISTER_USER03")
      val now = System.currentTimeMillis()

      // CREATE
      val outputRegisterUserRef01Ent = client.outputRegisterUser(null)
      var outputRegisterUserRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.output_register_user"), "output_register_user_ref01"))
      val outputRegisterUserRef01DataResult = outputRegisterUserRef01Ent.create(outputRegisterUserRef01Data, null)
      outputRegisterUserRef01Data = Helpers.toMapAny(outputRegisterUserRef01DataResult)
      rep.check("output_register_user.create.map", outputRegisterUserRef01Data != null, "expected create result to be a map")
    }
  }
}

File: elixir/test/output_register_user_entity_test.exs
# OutputRegisterUser entity test (offline, mock transport)

defmodule BluefinTecsUserBackoffice.OutputRegisterUserEntityTest do
  use ExUnit.Case

  alias Voxgig.Struct, as: S
  alias BluefinTecsUserBackoffice.Helpers, as: H
  alias BluefinTecsUserBackoffice.Json

  defp fixture do
    Json.parse(File.read!("../.sdk/test/entity/output_register_user/OutputRegisterUserTestData.json"))
  end

  defp mk_sdk do
    existing = H.or_(S.getpath(fixture(), "existing"), S.jm([]))
    BluefinTecsUserBackoffice.test(S.jm(["entity", existing]))
  end

  defp first_id do
    existing = H.or_(S.getpath(fixture(), "existing.output_register_user"), S.jm([]))
    keys = S.keysof(existing)
    if keys == [], do: nil, else: hd(keys)
  end

  test "should create instance" do
    sdk = BluefinTecsUserBackoffice.test()
    ent = BluefinTecsUserBackoffice.output_register_user(sdk)
    assert ent != nil
  end

  test "should create then read back" do
    sdk = BluefinTecsUserBackoffice.test(S.jm(["entity", S.jm(["output_register_user", S.jm([])])]))
    ent = BluefinTecsUserBackoffice.output_register_user(sdk)
    made = BluefinTecsUserBackoffice.Entity.OutputRegisterUser.create(ent, S.jm(["name", "test-create"]))
    assert S.ismap(made)
    assert S.getprop(made, "id") != nil
  end
end
